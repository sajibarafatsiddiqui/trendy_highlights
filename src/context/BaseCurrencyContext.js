import React from 'react';

const BaseCurrencyContext = React.createContext();
export default BaseCurrencyContext;
