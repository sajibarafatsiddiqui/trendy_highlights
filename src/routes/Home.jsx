import BaseCard from 'components/baseCard';
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { fetchAllCurrencyAgainstBase } from 'redux/currencySlice';

const Home = () => {
  const dispatch = useDispatch();
  const baseCurrency = 'USD';
  const { rate } = useSelector( (state) => state.currency )
  useEffect( () => dispatch( fetchAllCurrencyAgainstBase( {baseCurrency}) ),[ baseCurrency,dispatch])
  return (
    <div>
        <BaseCard baseCurrency={ baseCurrency } />
        <Cards rate = {rate} />
    </div>
  )
}

export default Home