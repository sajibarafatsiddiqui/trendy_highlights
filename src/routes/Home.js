import BaseCard from 'components/BaseCard';
import Cards from 'components/Cards';
import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAllCurrencyAgainstBase } from 'redux/currencySlice';
import { pick } from 'lodash';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import Row from 'react-bootstrap/Row';
import { Card } from 'react-bootstrap';
import style from 'styles/CurrencyCard.module.css';

const Home = () => {
  const dispatch = useDispatch();
  const { baseCurrency } = useContext(BaseCurrencyContext);
  const { rates } = useSelector((state) => state.currency);
  const filteredRates = pick(rates,
    'AED', 'AUD', 'BDT', 'CAD', 'CHF', 'CNY', 'EUR',
    'GBP', 'INR', 'JPY', 'KWD', 'NZD', 'RMB');
  useEffect(() => {
    dispatch(fetchAllCurrencyAgainstBase(baseCurrency));
  }, [dispatch, baseCurrency]);
  return (
    <>
      <p className={style.dark_header}>Exchange Rate</p>
      <Row xs={1} md={1} className="g-0">
        <BaseCard baseCurrency={baseCurrency} />
      </Row>
      <Row xs={1} md={1} className="g-0">
        <Card.Text className={style.dark}>Exchange Rate By Currency</Card.Text>
      </Row>
      <Cards rate={filteredRates} />
    </>
  );
};

export default Home;
