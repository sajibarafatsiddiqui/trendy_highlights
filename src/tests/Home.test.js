import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import Home from 'routes/Home';
import { Provider } from 'react-redux';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import { BrowserRouter as Router } from 'react-router-dom';
import store from 'redux/store';


jest.mock('axios');
axios.get.mockResolvedValue({
    data: {
    response: {
        base: "USD",
        date: "2023-03-17T11:55:16Z",
        rates:{
        ADA: 2.91801541,
        AED: 3.6725,
        EUR: 87.69844742,
        ALL: 107.8995027,
        }
        },
}
});

const baseCurrency ='USD'

const wrap = (component) => (
    <Provider store={store}>
      <BaseCurrencyContext.Provider value={{ baseCurrency }}>
      <Router>
      {component}
      </Router>
      </BaseCurrencyContext.Provider>
    </Provider>
);

describe('Tests for <Home /> component', () => {
  render(wrap(<Home />));
  it('Should render childs from API correctly', () => {
    expect(screen.getByText('AED')).not.toBeNull();
  });

  it('Should display currencies as links', () => {
    render(wrap(<Home />));
    expect(screen.getByText('AED').closest('a').href.includes('AED')).toBeTruthy();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(<Home />)).toJSON()).toMatchSnapshot();
  });
});