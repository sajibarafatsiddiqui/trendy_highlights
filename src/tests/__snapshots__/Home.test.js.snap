// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Home /> component Should match the snapshot 1`] = `
Array [
  <p
    className="dark_header"
  >
    Exchange Rate
  </p>,
  <div
    className="g-0 row row-cols-md-1 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="USD BaseCard card"
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          />
          <p
            className="card-text"
          />
          <div
            className="card-footer"
          >
            <h1>
              USD
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="g-0 row row-cols-md-1 row-cols-1"
  >
    <p
      className="dark card-text"
    >
      Exchange Rate By Currency
    </p>
  </div>,
  <div
    className="g-0 row row-cols-md-2 row-cols-2"
  >
    <div
      className="col"
    >
      <div
        className="AED Card card"
        style={
          Object {
            "backgroundColor": "#CF4277",
            "padding": "0",
          }
        }
      >
        <div
          className="card-body"
          style={
            Object {
              "margin": "0",
            }
          }
        >
          <a
            className=""
            href="/details/USD-AED"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
                "textDecoration": "none",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "position": "absolute",
                  "right": "0",
                  "top": "0",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M262.62 336L342 256l-79.38-80m68.35 80H170"
                fill="none"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="32"
              />
              <path
                d="M256 448c106 0 192-86 192-192S362 64 256 64 64 150 64 256s86 192 192 192z"
                fill="none"
                strokeMiterlimit="10"
                strokeWidth="32"
              />
            </svg>
            <div
              className="card-title h5"
            />
            <p
              className="card-text"
            >
              AED
            </p>
            <div
              className="card-footer"
              style={
                Object {
                  "bottom": "0",
                  "position": "absolute",
                  "right": "0",
                }
              }
            >
              3.6725
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="EUR Card card"
        style={
          Object {
            "backgroundColor": "#e94987",
            "padding": "0",
          }
        }
      >
        <div
          className="card-body"
          style={
            Object {
              "margin": "0",
            }
          }
        >
          <a
            className=""
            href="/details/USD-EUR"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
                "textDecoration": "none",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "position": "absolute",
                  "right": "0",
                  "top": "0",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M262.62 336L342 256l-79.38-80m68.35 80H170"
                fill="none"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="32"
              />
              <path
                d="M256 448c106 0 192-86 192-192S362 64 256 64 64 150 64 256s86 192 192 192z"
                fill="none"
                strokeMiterlimit="10"
                strokeWidth="32"
              />
            </svg>
            <div
              className="card-title h5"
            />
            <p
              className="card-text"
            >
              EUR
            </p>
            <div
              className="card-footer"
              style={
                Object {
                  "bottom": "0",
                  "position": "absolute",
                  "right": "0",
                }
              }
            >
              87.6984
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
